### PR TITLE
chore: Add ListPage Proposal

### PR DESCRIPTION
**Pull request by seanhealy**
### Added

- Proposal for a `ListView` component.

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: docs/proposals/ListView.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Component Name
+
+The `ListPage` component will provide the structure and parts needed to display
```
Comment (eddysims) [inline docs/proposals/ListView.md:3]: `ListPage` or `ListView`? the title of the PR is `ListView`

File: docs/proposals/ListView.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+a list view.
+
+## Design Patterns
+
+There is a desire to provide developers with page templates to both speed up the
+development of pages but also provide our users with a more consistent
+experience.
```
Comment (eddysims) [inline docs/proposals/ListView.md:10]: I wonder if Atlantis is the right place for this? I understand that we want this component and I see the value in it, but is there a risk in providing a ListPage layout to an open source project? Maybe there is a better place for this component?

File: docs/proposals/ListView.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+} from "@jobber/…"
+
+<ListPage
+  title="Page Title"
+  intro="Description of this page."
+  actions={[ButtonProps]}
+  filters={} // optional
+  searchable={true} // optional
+
+  pageSize={50} // optional
```
Comment (eddysims) [inline docs/proposals/ListView.md:120]: What would this prop do? This one seems unclear to me.

File: docs/proposals/ListView.md (hunk 1 above)
Comment (seanhealy) [reply docs/proposals/ListView.md:3]: I'm not sure. Do you have any thoughts on that? I started with View then realized it sort of is a whole page…?

File: docs/proposals/ListView.md (hunk 3 above)
Comment (seanhealy) [reply docs/proposals/ListView.md:120]: This was a pondering that would let you change the records to fetch per page.

**Comment by eddysims** [comment]
I wonder if Atlantis is the right place for this component? I understand that we want this component and I see the value in it, but is there a risk in providing a ListPage layout to an open source project? Maybe there is a better place for this component?

File: docs/proposals/ListView.md (hunk 3 above)
Comment (seanhealy) [reply docs/proposals/ListView.md]: Discussed this and pulled it out for now.

File: docs/proposals/ListView.md (hunk 1 above)
Comment (seanhealy) [reply docs/proposals/ListView.md:3]: Moved things to ListPage after talking.

Comment (JeffMarvin) [reply docs/proposals/ListView.md:3]: ^ This proposal file needs a rename, then.

File: docs/proposals/ListView.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+## Design Patterns
+
+There is a desire to provide developers with page templates to both speed up the
+development of pages but also provide our users with a more consistent
+experience.
+
+A first step in this is imagining what a common `ListPage` component would look
+like. It would need to have some level of flexibility to allow for use on
+different pages but still provide common "slots" or areas where.
+
```
Comment (JeffMarvin) [inline docs/proposals/ListView.md:15]: Is it worth mentioning here how this pattern should be addressed over time? I think we're starting simple here, which is good. 
But a proposal for a general layout template approach like this should probably be discussed or considered by a number of people... how do we work with designers and developers to get their input on this?

File: docs/proposals/ListView.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+## Accessibility
+
+As a primarily layout component we don't have a lot of interaction concerns on
+the ListPage but, we should ensure that we're using semantically appropriate
+tags wherever possible to aid in readability and navigation.
+
+## Responsiveness & Mobile
+
```
Comment (JeffMarvin) [inline docs/proposals/ListView.md:23]: What is taking care of the page title and the action buttons? How do they relate to this component, aren't they on the page?

File: docs/proposals/ListView.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+https://www.figma.com/file/qw8HnK3rBDHGf44RVksCXT/JOB-20476-Manage-Team?node-id=139%3A245
+
+## Interface
+
+A decent number of these parts will move into a `DataTable` component after we
+have that RFC complete and some more experience trying this approach in this
+component.
+
+Should we include items as a list prop? We could nest a list view? How to handle
```
Comment (JeffMarvin) [inline docs/proposals/ListView.md:45]: There seems to be a few questions here, so separating them out more fully would be useful and more actionable.

File: docs/proposals/ListView.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+We may want a better name than column here. For the purposes of this Column
+means a section or cell that is common across each row or list item. For example
+"name" or "actions".
+
+```tsx
+interface ListColumn {
+  key: string;
+  // A Renderer such as the CustomCell renderer.
+  // Built in renderers will also be provided.
```
Comment (JeffMarvin) [inline docs/proposals/ListView.md:58]: Can you expand on this? Seems big.

File: docs/proposals/ListView.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+interface TableColumn extends ListColumn {
+  header: string;
+}
+```
+
+### CustomListRenderer
+
+A List renderer would conform to the following.
```
Comment (JeffMarvin) [inline docs/proposals/ListView.md:70]: There seems to be very little explaining what these components are and how they work together. Would it be possible to expand a bit on that? This document just drops into chunks of code with no explanation.

File: docs/proposals/ListView.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+}
+```
+
+```tsx
+import {
+  ListPage,
+  TableListRenderer,
+  TextCell,
+} from "@jobber/…"
+
```
Comment (JeffMarvin) [inline docs/proposals/ListView.md:112]: I see how the layout is achieved but this solution seems fairly tied to the data side of the component. Is there a more generalized version of that we should try for here?

**Comment by JeffMarvin** [comment]
Had some thoughts, and also am curious what your response to Eddy's comment here (https://github.com/GetJobber/atlantis/pull/233#issuecomment-618517288) is. I imagine this is actually a proposal for a Jobber component.

File: docs/proposals/ListView.md (hunk 5 above)
Comment (seanhealy) [reply docs/proposals/ListView.md:23]: Looked at this in the interface section.

File: docs/proposals/ListView.md (hunk 6 above)
Comment (seanhealy) [reply docs/proposals/ListView.md:45]: This was an old line if forgot to remove sorry!

File: docs/proposals/ListView.md (hunk 8 above)
Comment (seanhealy) [reply docs/proposals/ListView.md:70]: Wanna find a couple minutes to chat so I can get a better idea what you're wondering about here? 😄

File: docs/proposals/ListView.md (hunk 9 above)
Comment (seanhealy) [reply docs/proposals/ListView.md:112]: I don't quite follow… Sorry.

Comment (seanhealy) [reply docs/proposals/ListView.md]: Discussed on the ol' internet of voice.

**Comment by JeffMarvin** [comment]
@eddysims I'm good with everything here now. You had a "changes requested" review on this, so I'll leave it up to you to pass it when its good!

**Comment by eddysims** [comment]
LGTM

**Pull request merged**